server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -345,6 +345,12 @@ public ClusterState applyCreateIndexRequest(ClusterState currentState, CreateInd
                 final List<IndexTemplateMetadata> v1Templates = MetadataIndexTemplateService.findV1Templates(currentState.metadata(),
                     request.index(), isHiddenFromRequest);
 
+                if (v1Templates.size() > 1) {
+                    deprecationLogger.deprecatedAndMaybeLog("index_template_multiple_match", "index [{}] matches multiple templates " +
+                        "[{}], and won't be supported with v2 index templates", request.index(),
```

Comment:
I think this is a little confusing about what will not be supported and what is not going to be supported going forward, maybe it should be something like:

"index [foo] matches multiple v1 templates [t1, t2], v2 index templates will only match a single index template"

What do you think?

Reply:
This is good idea, I've changed it